include/sway/input/keyboard.h

```diff
@@ -3,7 +3,13 @@
 
 #include "sway/input/seat.h"
 
-#define SWAY_KEYBOARD_PRESSED_KEYSYMS_CAP 32
+#define SWAY_KEYBOARD_PRESSED_KEYS_CAP 32
+
+struct sway_shortcut_state {
+	uint32_t pressed_keys[SWAY_KEYBOARD_PRESSED_KEYS_CAP];
+	uint32_t pressed_keycodes[SWAY_KEYBOARD_PRESSED_KEYS_CAP];
```

Comment:
Why are the keycodes duplicated in each state?

Reply:
They are necessary to keep track of the keysyms corresponding to a given keycode. (This is cleaner than trying to reverse the xkb API to find what the keysym would have been at the time the corresponding keycode was pressed; consider the shortcut `Ctrl+b+at` in which `Shift` is pressed halfway through.)

They are somewhat duplicated (up to translation by 8) for the shortcut model tracking keycodes, although even in this case, after a modifier key is erased, `pressed_keys[i]` might be `0` while `pressed_keycodes[i]` stores the event keycode.

Reply:
Oh, right, that makes a lot more sense. Can you add a comment explaining the mapping between these two guys?